UNITTESTS/stubs/CMakeLists.txt

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+add_library(mbed-os-stubs-headers INTERFACE)
```

Comment:
if this is interface, why others are static libs ? `add_library(mbed-os-stubs-hal)` is static libray, isn't it?

Reply:
The idea is to build all the stub library once as a static lib, and reuse across unittest test suite wherever dependent on hal stub

Reply:
so this should be as well STATIC

Reply:
interface library doesn't build anything - which makes sense for header only libraries

Reply:
correct, this is headers lib only.

Reply:
that is my point: use of INTERFACE for a header only lib is wholly appropriate